djproxy/views.py

```diff
@@ -135,6 +135,10 @@ def proxy(self):
         headers['X-Forwarded-Host'] = self.request.get_host()
         headers['X-Forwarded-For'] = self.xff
 
+        for prefix, url in self.reverse_urls:
+            if url == self.proxy_url and self.request.path.startswith(prefix):
```

Comment:
`proxy_url` requires information from the route to work correctly. It can't just take the path from the request and mash it onto the upstream base url. It needs to know what subset of the request path to mash onto the url (which is typically achieved with a named pattern).

In your test, `proxy_url` will always equal the base url unless you specific a `url` kwarg to the view. The url `kwarg` is appended to the `base_url`